multiqc/core/ai.py

```diff
@@ -0,0 +1,764 @@
+                },
+            )
+
+        if config.verbose:
+            response = send_request()
+        else:
+            response = run_with_spinner("ai", "Summarizing report with AI...", send_request)
+
+        if response.status_code != 200:
+            msg = f"Failed to get a response from Seqera: {response.status_code} {response.text}"
+            logger.error(msg)
+            if config.strict:
+                raise RuntimeError(msg)
+            return None
+
+        response_dict = response.json()
+        uuid = response_dict.get("uuid")
+        generation: Dict[str, str] = response_dict.get("generation")
+        return InterpretationResponse(
+            uuid=uuid,
+            interpretation=InterpretationOutput(**generation),
+            model=self.model or "claude-3-5-sonnet-latest",
+        )
+
+
+def get_llm_client() -> Optional[Client]:
+    if not config.ai_summary:
+        return None
+
+    if config.ai_provider == "seqera":
+        api_key = os.environ.get("SEQERA_ACCESS_TOKEN", os.environ.get("TOWER_ACCESS_TOKEN"))
+        if not api_key:
+            logger.error(
+                "config.ai_summary is set to true, and config.ai_provider is set to 'seqera', "
+                "but Seqera access token is not set. "
+                "Please set the SEQERA_ACCESS_TOKEN / TOWER_ACCESS_TOKEN environment variable "
+                "or change config.ai_provider"
+            )
+            return None
+        return SeqeraClient(config.ai_model, api_key)
+
+    if config.ai_provider == "anthropic":
+        api_key = os.environ.get("ANTHROPIC_API_KEY")
+        if not api_key:
+            logger.error(
+                "config.ai_summary is set to true, and config.ai_provider is set to 'anthropic', but Anthropic API "
+                "key not set. Please set the ANTHROPIC_API_KEY environment variable, or change config.ai_provider"
+            )
+            return None
+        try:
+            return AnthropicClient(api_key)
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "AI summary requested through `config.ai_summary`, but required dependencies are not installed. Install them with `pip install multiqc[anthropic]`"
+            )
+
+    if config.ai_provider == "openai":
+        api_key = os.environ.get("OPENAI_API_KEY")
+        if not api_key:
+            logger.error(
+                "config.ai_summary is set to true, and config.ai_provider is set to 'openai', but OpenAI API "
+                "key not set. Please set the OPENAI_API_KEY environment variable, or change config.ai_provider"
+            )
+            return None
+        try:
+            return OpenAiClient(api_key)
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "AI summary requested through `config.ai_summary`, but required dependencies are not installed. Install them with `pip install multiqc[openai]`"
+            )
+
+    return None
```

Comment:
Do we want to do an `else` here and log an error? Or are we confident that upstream validation will mean this can never happen?

Reply:
Good catch, thank you! It would validate CLI, but not the YAML config. Updated